### ISSUE TITLE
Modelo Entrenamiento en Python

### ISSUE THREAD

**Issue by Fhzh29** [Fhzh29/IAUAC#1]
import numpy as np

# Parámetros del algoritmo
learning_rate = 0.1  # coeficiente de aprendizaje (α)
threshold = 0.01  # umbral de error aceptable
epochs = 1000  # número máximo de iteraciones

# Inicialización de pesos y umbral (θ) en valores pequeños aleatorios
np.random.seed(0)  # para reproducibilidad
weights = np.random.uniform(-1, 1, size=(3,))  # Asumimos una entrada de tres características
bias = np.random.uniform(-1, 1)  # umbral o bias

# Función de activación (función escalón)
def activation_function(x):
    return 1 if x >= 0 else 0

# Entrenamiento del algoritmo
def train(X, D):
    global weights, bias
    for epoch in range(epochs):
        total_error = 0
        for i in range(len(X)):
            # Propagación hacia adelante: cálculo de la salida y
            net_input = np.dot(X[i], weights) - bias
            y = activation_function(net_input)
            
            # Cálculo del error
            error = D[i] - y
            total_error += abs(error)
            
            # Retropropagación: ajuste de los pesos y el umbral
            weights += learning_rate * error * X[i]
            bias -= learning_rate * error
        
        # Verificación de condición de parada
        if total_error < threshold:
            print(f"Entrenamiento completado en época {epoch + 1}")
            break
    else:
        print("No se alcanzó un error aceptable después de la cantidad máxima de épocas.")

# Datos de ejemplo
# X: conjunto de entrenamiento con dos características más un término de sesgo (1, x1, x2)
# D: salidas deseadas
X = np.array([[1, 0, 0], [1, 0, 1], [1, 1, 0], [1, 1, 1]])  # incluye el sesgo
D = np.array([0, 0, 0, 1])  # salida deseada (AND lógico)

# Entrenar el modelo
train(X, D)

# Mostrar los pesos finales
print("Pesos finales:", weights)
print("Umbral final:", bias)
